amqp/dotnet/SharedClass.cs

```diff
@@ -0,0 +1,144 @@
+
+        public void PutMessages(string symbolType)
+        {
+            string add = "amqp://" + Username + ":" + Password + "@" + hostName + ":" + port;
+            Address address = new Address(add);
+
+            // Create Connection
+            Connection connection = new Connection(address);
+
+            // Create Session
+            Session session = new Session(connection);
+
+            // Specifies the endpoint as QUEUE/TOPIC
+            Target target = new Target();
+            target.Address = symbolName;
+            target.Capabilities = new Symbol[]{
+                new Symbol(symbolType)
+            };
+
+            void OnAttached(ILink link, Attach attach)
+            {
+                // Handle the attachment event
+                if (attach != null && target.Address != null)
+                    Console.WriteLine("Sender link attached successfully!");
+                else
+                    Console.WriteLine("Sender link attachment failed!");
+            }
+
+            // Create SenderLink
+            SenderLink sender = new SenderLink(session, "client", target, OnAttached);
+
+            // Create Message Object pointing to endpoint
+            Message message = new Message("Hello AMQP!");
+            message.Properties = new Properties();
+            message.Properties.To = add;
+
+            // put/publish the message to specific queue/topic
+            sender.Send(message);
+            Console.WriteLine("message sent < " + message.Body.ToString() + " >.. ");
+            Console.WriteLine("Put/Pub Successfull");
+
+            // close the connection
+            sender.Close();
+            session.Close();
+            connection.Close();
+        }
+
+        public void GetMessages(string symbolType)
+        {
+            string add = "amqp://" + Username + ":" + Password + "@" + hostName + ":" + port;
+            Address address = new Address(add);
+
+            // Create Connection
+            Connection connection = new Connection(address);
+
+            //Create Session
+            Session session = new Session(connection);
+
+            // Specifies the endpoint as QUEUE/TOPIC
+            Symbol[] s = new Symbol[]{
+                new Symbol(symbolType)
+            };
+            // Create Source
+            Source source = new Source();
+            source.Address = symbolName;
+            source.Capabilities = s;
+
+            void OnAttached(ILink link, Attach attach)
+            {
+                // Handle the attachment event
+                if (attach != null && source.Address != null)
+                    Console.WriteLine("Receiver link attached successfully!");
+                else
+                    Console.WriteLine("Receiver link attachment failed!");
+            }
+
+            // Create ReceiverLink
+            ReceiverLink receiver = new ReceiverLink(session, "client", source, OnAttached);
+
+            while (true)
```

Comment:
This looks like a tight loop. Is it possible for receiver to get with wait to allow time for a message to appear from a producer? Two instances of the application could be run individually.

Reply:
The receiver will wait for 60 seconds, and if no action occurs within that time, the connection will be closed.